odoo/tools/pdf.py

```diff
@@ -3,33 +3,27 @@
 
 import io
 
-from pikepdf import Pdf, AttachedFileSpec
+from pikepdf import Pdf
 
 
 class OdooPdf(Pdf):
-    def new(*args, **kwargs) -> 'OdooPdf':
-        pdf = Pdf.new(*args, **kwargs)
-        pdf.__class__ = OdooPdf
-        return pdf
-
-    def open(*args, **kwargs) -> 'OdooPdf':
-        pdf = Pdf.open(*args, **kwargs)
-        pdf.__class__ = OdooPdf
-        return pdf
-
-    def save(self, *args, **kwargs):
+    def _save(self, *args, **kwargs):
         with self.open_metadata(set_pikepdf_as_editor=False) as meta:
             meta['xmp:CreatorTool'] = 'Odoo'
             meta['pdf:Producer'] = "Odoo"
-        super().save(*args, **kwargs)
+        self.save_bak(*args, **kwargs)
 
-    def get_attachments(self):
+    def _get_attachments(self):
         for key, value in self.attachments.items():
             yield (key, value.get_file().read_bytes())
 
-    def add_attachment(self, fname, fdata):
-        self.attachments[fname] = AttachedFileSpec(self, data=fdata)
+    def _add_attachment(self, fname, fdata):
+        self.attachments[fname] = self._attach(basename=fname, filebytes=fdata)
 
+    Pdf.add_attachment = _add_attachment
+    Pdf.get_attachments = _get_attachments
+    Pdf.save_bak = Pdf.save
+    Pdf.save = _save
```

Comment:
![this-isnt-your-everyday-evil-this-is-advanced-evil](https://user-images.githubusercontent.com/7571158/135419232-43a70564-39a8-4e26-ab47-8f81a39b5e25.jpg)